### ISSUE TITLE
Manifest

### ISSUE THREAD

**Issue by jackdaniels552** [pvrs12/Anesidora#104]
Manifest is on v2 and needs update v3 due to changes in 2023. Cant load due to manifest version 2.

**Comment by Bolongna** [comment]
Yeah it would be nice to see an update that will run on V3 but I'm not sure just what would need to be changed to keep it working.

**Comment by hucario** [comment]
Hi, sorry for the late response!
We actually discussed this in #93 .

In short, it's unfeasible because we open a "background page" - think a hidden tab - to play the music from. Manifest v3 completely removes background pages in favor of short-lived event listener pages. There are some alternatives available, but I don't believe they're worth pursuing:
1. Chrome offers a nonstandard API to create a background page. It adds a lot of friction compared to the MV2 method (you need to provide a reason and justification for having a background page, among other things), but it's doable and would be a good solution. The main problem with this is that it's chrome-only, meaning that I'd have to maintain a Firefox version and a Chrome version separately.
2. We *could* hijack an active tab to play our audio from. This would technically fulfill our requirement to not add new tabs, but it would add so much friction and so many bugs that it would be a bad experience and I'm willing to outright dismiss it. Off the top of my head, if you closed the host tab it would, at minimum, add a gap in the playback. You would be unable to mute the tab without also muting the music. It's just not great.
3. We could drop the "tabless" requirement and just have it be another tab. This is the most reasonable option, and I may do it in the future.

The main problem is that only Chrome requires MV3, and we're banned from the Chrome Web Store due to DMCA/copyright reasons. So any method we use would only be for users who are willing to enter chrome's extension development mode and load unpacked, in which case you can just launch with `--enable-features=AllowLegacyMV2Extensions --disable-features=ExtensionManifestV2Unsupported,ExtensionManifestV2Disabled,ExtensionsManifestV3Only` and use our MV2 version. Firefox allows MV2 just fine, so this isn't an issue there.

This all taken into account, MV3 is currently not a priority, especially given my small amount of time for developing new features. Sorry! If you really do want mv3 and are willing to deal with another open tab, I would be willing to develop an MV3 version in the future - but it would take a while.
